plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -705,22 +703,60 @@ public boolean downloadPlugin(Plugin plugin, File location) {
         }
         String pluginDownloadUrl = getPluginDownloadUrl(plugin);
         boolean successfulDownload = downloadToFile(pluginDownloadUrl, plugin, location);
-        if (!successfulDownload) {
-            logVerbose(String.format("First download attempt of %s unsuccessful, reattempting",
-                    plugin.getName()));
-            //some plugin don't follow the rules about artifact ID, i.e. docker-plugin
+
+        //try to set an alternate version in case requested version and version in json don't exactly match
+        if (!successfulDownload && setAlternateVersion(plugin)) {
+            logVerbose(String.format("Download attempt of %s unsuccessful, reattempting", plugin.getName()));
+            pluginDownloadUrl = getPluginDownloadUrl(plugin);
+            successfulDownload = downloadToFile(pluginDownloadUrl, plugin, location);
+        }
+
+        //some plugin don't follow the rules about artifact ID, i.e. docker-plugin
+        if (!successfulDownload && (plugin.getName().equals(plugin.getOriginalName()))) {
+            logVerbose(String.format("Download attempt of %s unsuccessful, reattempting", plugin.getName()));
             String newPluginName = plugin.getName() + "-plugin";
             plugin.setName(newPluginName);
             pluginDownloadUrl = getPluginDownloadUrl(plugin);
             successfulDownload = downloadToFile(pluginDownloadUrl, plugin, location);
+                if (!successfulDownload && setAlternateVersion(plugin)) {
+                    logVerbose(String.format("Download attempt of %s unsuccessful, reattempting", plugin.getName()));
+                    pluginDownloadUrl = getPluginDownloadUrl(plugin);
+                    successfulDownload = downloadToFile(pluginDownloadUrl, plugin, location);
+                }
         }
+
         if (successfulDownload && location == null) {
             System.out.println(String.format("%s downloaded successfully", plugin.getName()));
             installedPluginVersions.put(plugin.getName(), pluginVersion);
         }
         return successfulDownload;
     }
 
+    /**
+     * Sets an alternate but equivalent plugin version by either appending or deleting a patch of 0. Filters out
+     * incremental plugins and plugins of formats like the following: 4.4.4-3.0"
+     *
+     * @param plugin plugin to set
+     * @return
+     */
+    public boolean setAlternateVersion(Plugin plugin) {
```

Comment:
why are we doing this? I assume dependencies will always have a correct version, and we shouldn't fix user errors silently? it should just fail and let the user know what plugin was wrong

Reply:
You would think the dependencies would have the right version, but I had previously found an example where this wasn't the case. The vsphere-cloud plugin version says it requires 1.5, but the plugin-versions.json has dependency information for 1.5.0 and 1.5.0 is also the version on the download page.  I'm looking at it now and it seems like you can download the plugin if 1.5 or 1.5.0 is listed as the version, but I thought that this wasn't working previously. In this case, since the user didn't directly request that plugin (it was just a dependency), so if it failed it wouldn't be their fault.

Reply:
metadata looks fine to me?
https://updates.jenkins.io/current/update-center.actual.json
this seems like a giant hack and we should fix the plugin if we can find the issue

Reply:
The metadata for this plugin is still messed up. Vsphere-cloud says it requires node-iterator-api version 1.5, even though that version doesn't exist for node-iterator-api here: http://updates.jenkins.io/download/plugins/node-iterator-api/ or here:  http://mirror.xmission.com/jenkins/updates/current/plugin-versions.json, although apparently it doesn't truly matter in this case since 
http://updates.jenkins.io/download/plugins/node-iterator-api/1.5.0/node-iterator-api.hpi and
http://updates.jenkins.io/download/plugins/node-iterator-api/1.5/node-iterator-api.hpi will both download the plugin.  

However, in the cases where there's a version mismatch of a dependency and the plugin can't be downloaded, with the newer PRs an error would be thrown immediately if the plugin dependencies can't be resolved (if the plugin dependencies can't be determined from plugin-versions.json, which is dependent on having the right plugin version, *or* the plugin's dependencies are determined from the manifest, which is still dependent on being able to download the plugin, which is dependent on having the right version to build the download url) and there'd be nothing the user could do. 

I see what you're saying though, and this could potentially cause more problems than it solves.  I didn't really have any idea of how common this issue might be, but maybe it's pretty rare.  I'll close this PR and we can see if it's actually a problem users encounter. I'm also open to suggestions on other ways to address this.

Reply:
Looks like both versions are in artifactory:
https://repo.jenkins-ci.org/webapp/#/search/package/eyJzZWFyY2giOiJxdWljayIsInF1ZXJ5Ijoibm9kZS1pdGVyYXRvci1hcGkifQ==
Probably a failed release, sounds like a good idea to hold off on this change 👍